File: 322-coin-change/coin-change.py
class Solution:
    def coinChange(self, coins: List[int], amount: int) -> int:
        index = [float(inf)]* (amount+1)
        index[0]=0

        for c in coins:
            for i in range(c,amount+1):
                if i-c>=0:
                    index[i] = min(index[i],index[i-c]+1)
        
        return index[amount] if index[amount] != float(inf) else -1

File: 322-coin-change/coin-change.java
class Solution {
    public int coinChange(int[] coins, int amount) {
        int[] arr = new int[amount+1];
        Arrays.fill(arr, Integer.MAX_VALUE);
        arr[0] = 0;
        for(int i = 1; i<amount+1; i++){
            for(int c: coins){
                
                if(i-c>=0 && arr[i - c] != Integer.MAX_VALUE){
                    arr[i] = Math.min(arr[i],arr[i-c]+1);
                }
            }
        }
        return arr[amount] != Integer.MAX_VALUE ? arr[amount] : -1;
        
    }
}